llvm/lib/Target/DirectX/DXILOpLowering.cpp

```diff
@@ -463,6 +463,28 @@ class OpLowerer {
     });
   }
 
+  [[nodiscard]] bool lowerUpdateCounter(Function &F) {
```

Comment:
Sorry for the late review. Is this going to be a common pattern to create a handle and then use it as an arg? Maybe it is possible to make a general helper.

Reply:
Most of the intrinsics that use a handle are related to loads, stores, and sampling - bufferUpdateCounter is a bit special in its simplicity. As you might see from the handling of typed buffer loads and stores, the translation isn't as straightforward there. @joaosaffran asked me about generalizing this now, but I think it's probably best to just do this one directly for now and generalize when we have more related logic in to make sure we're not generalizing in a way we'll need to undo later.

Reply:
Okay, sounds good 👍